### ISSUE TITLE
YAML-references-file generated by Pandoc not acceptable for Pandoc --citeproc

### ISSUE THREAD

**Issue by jkrenzer** [jgm/pandoc#7708]
**Explain the problem.**
The yaml-formatted reference-files generated by pandoc are not accepted by pandoc --citeproc as bibliographies anymore.

How to reproduce:

Make a `bibliography.yaml` with the content:

```yaml
---
references:
- author:
  - family: Krenzer
    given: Jörn
  container-title: Pandoc Issues
  id: foo-bar
  issued: 2021
  language: en-EN
  number: 0815
  publisher: A.C.M.E. Coporation
  title: Who did break my yaml?!
  type: paper-conference
  venue: Internet
...
```

Make a example markdown file `test.md`:

```markdown
---
title: References
nocite: @*
---

::: {#refs}
:::
```

Pandoc --citeproc will succeed with this file:

```bash
pandoc -f markdown -t html -s --bibliography bibliography.yaml --citeproc -o bibliography.html test.md
```

Now round-trip the yaml through pandoc. It also works when converting to e.g. biblatex and back to yaml.

```bash
pandoc -s -f markdown -t markdown -o bibliography_parsed.yaml bibliography.yaml
```

Now repeat the command given above with the pandoc-generated `bibliography_parsed.yaml` to reproduce the error:

```bash
pandoc -f markdown -t html -s --bibliography bibliography_parsed.yaml --citeproc -o bibliography.html test.md
Multiple YAML documents encountered
```
The error goes away, when the `---` set by pandoc as the end of the yaml-block are changed to `...`. Might be a change in the upstream YAML-libraries tolerance for this. Tried with an old pandoc 2.11 the error *did not* occure.

**Pandoc version?**
 ```bash
pandoc 2.16.1
Compiled with pandoc-types 1.22.1, texmath 0.12.3.2, skylighting 0.12.1,
citeproc 0.6, ipynb 0.1.0.2
User data directory: /root/.local/share/pandoc
Copyright (C) 2006-2021 John MacFarlane. Web:  https://pandoc.org
This is free software; see the source for copying conditions. There is no
warranty, not even for merchantability or fitness for a particular purpose.
```

**Workarounds**

Manually or stream-edit the bibliography-files generated by pandoc to end the YAML-Block with `...`

**Comment by jgm** [comment]
I noticed this yesterday and fixed it in the 2.16.2 release!

**Comment by jkrenzer** [comment]
Super. Thank you! Then we're all set I guess.
